models/data_model_pivot.go

```diff
@@ -0,0 +1,114 @@
+package models
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/cockroachdb/errors"
+)
+
+type PivotMetadata struct {
+	Id             string
+	CreatedAt      time.Time
+	OrganizationId string
+
+	BaseTableId string
+	FieldId     *string
+	PathLinkIds []string
+}
+
+type Pivot struct {
+	Id             string
+	CreatedAt      time.Time
+	OrganizationId string
+
+	BaseTable    string
+	BaseTableId  string
+	PivotTable   string
+	PivotTableId string
+
+	Field   string
+	FieldId string
+
+	PathLinks   []string
+	PathLinkIds []string
+}
+
+func AdaptPivot(pivotMeta PivotMetadata, dataModel DataModel) Pivot {
+	pivot := Pivot{
+		Id:             pivotMeta.Id,
+		CreatedAt:      pivotMeta.CreatedAt,
+		OrganizationId: pivotMeta.OrganizationId,
+
+		BaseTableId: pivotMeta.BaseTableId,
+		PathLinkIds: pivotMeta.PathLinkIds,
+	}
+
+	baseTable := dataModel.AllTablesAsMap()[pivotMeta.BaseTableId]
+	pivot.BaseTable = baseTable.Name
+
+	if pivotMeta.FieldId != nil {
+		field := dataModel.AllFieldsAsMap()[*pivotMeta.FieldId]
+		pivot.Field = field.Name
+		pivot.FieldId = field.ID
+		// in this case, the pivot table is the base table
+		pivot.PivotTable = baseTable.Name
+		pivot.PivotTableId = baseTable.ID
+	} else {
+		field, _ := FieldFromPath(dataModel, pivot.PathLinkIds, pivot.BaseTable)
```

Comment:
You know I'm not fan of this 😛 
but I understand the validation check should failed early here

Reply:
Hmm yea, I just basically didn't want to write the function twice (once for validation and once to return the last field). I think this is legit